CMakeLists.txt

```diff
@@ -16,7 +16,29 @@ endif()
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
+#-------------------------------------------------------------------------------
+# Project customization. We have various tools which will write gitignored
+# .cmake files to the source tree which configure environment specific
+# default settings for advanced cases. We check for these and load them here.
+# Because this comes after project() declaration and before options, it can
+# change defaults in ways that can typically only be done from the command line.
+# This lets us simulate the effect of CMAKE_PROJECT_TOP_LEVEL_INCLUDES but
+# hard-coded for our project based on the availability of certain config
+# files.
+#-------------------------------------------------------------------------------
+
+macro(iree_include_optional_customization file_name)
+  if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file_name})
+    message(STATUS "Including customization file: ${file_name}")
+    include(${CMAKE_CURRENT_SOURCE_DIR}/${file_name})
+  endif()
+endmacro()
+
+iree_include_optional_customization(".toolchain_config.cmake")
+iree_include_optional_customization(".llvm_dep_config.cmake")
```

Comment:
I wonder if we can use presets here (https://github.com/openxla/iree/issues/5804, https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) instead of our own include mechanism

That would be more compatible with switching between setups and would integrate with IDEs: https://github.com/microsoft/vscode-cmake-tools/blob/main/docs/cmake-presets.md

We could have the script create a CMakeUserPresets.json, which inherits from CMakeDevPresets.json, for example